pkg/deploy/kubernetes_image_puller.go

```diff
@@ -501,7 +539,46 @@ func GetExpectedKubernetesImagePuller(ctx *DeployContext) *chev1alpha1.Kubernete
 	}
 }
 
-// Unisntall the CSV, OperatorGroup, Subscription, KubernetesImagePuller, and update the CheCluster to remove
+// UpdateDefaultImagesIfNeeded updates the default images from `spec.images` if needed
+func UpdateDefaultImagesIfNeeded(ctx *DeployContext) error {
+	specImages := StringToImageSlice(ctx.CheCluster.Spec.ImagePuller.Spec.Images)
+	defaultImages := GetDefaultImages()
+	if UpdateSpecImages(specImages, defaultImages) {
+		return SetImages(ctx, specImages)
+	}
+	return nil
+}
+
+// UpdateSpecImages returns true if the default images from `spec.images` were updated
+// with new defaults
+//
+// specImages contains the images in `spec.images`
+// defaultImages contains the current default images from the environment variables
+func UpdateSpecImages(specImages []ImageAndName, defaultImages []ImageAndName) bool {
+	match := false
+	for i, specImage := range specImages {
+		specImageName, specImageTag := util.GetImageNameAndTag(specImage.image)
+		for _, defaultImage := range defaultImages {
+			defaultImageName, defaultImageTag := util.GetImageNameAndTag(defaultImage.image)
+			// if the image tags are different for this image, then update
+			if defaultImageName == specImageName && defaultImageTag != specImageTag {
```

Comment:
What if the user wants to provide some own extra image except defaults? Should we update image puller spec? Or we are focusing on the pulling only the default images?

Reply:
will let @dkwon17 to comment, but the idea is that only the RELATED_IMAGES will be updated during the rollout update of a new version. If the admin has updated the spec with some custom image for pre-pulling it will be there after tthe rollout

Reply:
Yes, @ibuziuk is correct. The aim was to update only the defaults, while keeping user-provided images unchanged in the imagePuller spec.